versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java

```diff
@@ -668,6 +669,44 @@ private <ID, R> void bulkFetch(
       return;
     }
 
+    if (num <= MAX_PARALLEL_READS) {
+      bulkFetchSome(tableId, ids, r, keyGen, resultGen, notFound);
+    } else {
+      bulkFetchMany(tableId, ids, r, keyGen, resultGen, notFound);
+    }
+  }
+
+  private <ID, R> void bulkFetchSome(
+      String tableId,
+      ID[] ids,
+      R[] r,
+      Function<ID, ByteString> keyGen,
+      Function<Row, R> resultGen,
+      Consumer<ID> notFound)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    int num = ids.length;
+    @SuppressWarnings("unchecked")
+    ApiFuture<Row>[] handles = new ApiFuture[num];
+    for (int idx = 0; idx < ids.length; idx++) {
+      ID id = ids[idx];
+      if (id != null) {
+        ByteString key = keyGen.apply(id);
+        handles[idx] = backend.client().readRowAsync(tableId, key);
+      }
+    }
+
+    bulkFetchCollectResults(ids, r, resultGen, notFound, num, handles);
+  }
+
+  private <ID, R> void bulkFetchMany(
+      String tableId,
+      ID[] ids,
+      R[] r,
+      Function<ID, ByteString> keyGen,
+      Function<Row, R> resultGen,
+      Consumer<ID> notFound)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    int num = ids.length;
     @SuppressWarnings("unchecked")
     ApiFuture<Row>[] handles = new ApiFuture[num];
     int idx = 0;
```

Comment:
Not related to this change, but I'm failing to understand why we have two nested for-loops below? But maybe I hadn't enough coffee this morning :-)

Imho `MAX_BULK_READS` should be used like `MAX_BULK_MUTATIONS`, i.e., when creating the client:

```java
    stubSettings.bulkReadRowsSettings().setBatchingSettings(
        stubSettings.bulkReadRowsSettings().getBatchingSettings().toBuilder()
            .setElementCountThreshold((long) BigTableConstants.MAX_BULK_READS)
            .build());
```

Reply:
@snazy I can open a small PR for that, do you think it's worth it?

Reply:
Sure, go ahead